packages/web/server/queries/complex/concentrated-liquidity.ts

```
@@ -0,0 +1,526 @@
+          const priceRangePromise = Promise.all([
+            getClTickPrice({
+              tick: lowerTick,
+              baseAsset,
+              quoteAsset,
+            }),
+            getClTickPrice({
+              tick: upperTick,
+              baseAsset,
+              quoteAsset,
+            }),
+          ]);
+          const unclaimedRewardsPromise = getTotalClaimableRewards({
+            rawClaimableIncentiveRewards: claimable_incentives,
+            rawClaimableSpreadRewards: claimable_spread_rewards,
+          });
+          const userUnbondingPositionsPromise = getUserUnbondingPositions({
+            userOsmoAddress,
+          });
+          const delegatedPositionsPromise = getUserDelegatedClPositions({
+            userOsmoAddress,
+          });
+          const undelegatingPositionsPromise = getUserUndelegatingClPositions({
+            userOsmoAddress,
+          });
+          const stakeCurrencyPromise = getAsset({ anyDenom: "OSMO" });
+          const initialRangeAprPromise = getConcentratedRangePoolApr({
+            lowerTick: lowerTick.toString(),
+            upperTick: upperTick.toString(),
+            poolId: position.pool_id,
+          });
+
+          const [
+            priceRange,
+            unclaimedRewards,
+            userUnbondingPositions,
+            delegatedPositions,
+            undelegatingPositions,
+            stakeCurrency,
+            initialRangeApr,
+          ] = await Promise.all([
+            priceRangePromise,
+            unclaimedRewardsPromise,
+            userUnbondingPositionsPromise,
+            delegatedPositionsPromise,
+            undelegatingPositionsPromise,
+            stakeCurrencyPromise,
+            initialRangeAprPromise,
+          ]);
+
+          const pool = pools.find((pool) => pool.id === position.pool_id);
+
+          if (!pool) {
+            console.error(`Pool (${position.pool_id}) not found`);
+            return undefined;
```

Would prefer a throw here instead of error log.